pmd-java/src/main/resources/category/java/errorprone.xml

```diff
@@ -641,12 +641,14 @@ k = i * j;      // set k with 80 not 120
     <rule name="BadComparison"
```

Comment:
I think, it makes more sense to rename the rule as described here: https://pmd.github.io/latest/pmd_devdocs_rule_deprecation_policy.html#renaming-rules

You enhanced BadComparison in such a way, that the rule deserves a better name. But it is still detecting the same cases (wrong usages of NaN) - and even more, fixing false negatives.

I'll try this out locally when merging.